src/main/groovy/com/google/protobuf/gradle/ToolsLocator.groovy

```diff
@@ -108,4 +108,13 @@ class ToolsLocator {
       }
     }
   }
+
+  static List<String> artifactParts(String artifactCoordinate) {
+    final validName = '[+-.0-9A-Z_a-z]+'
+
+    ((artifactCoordinate =~ /(${validName}):(${validName}):(${validName})(:${validName})?(@${validName})?/)[0]
```

Comment:
I wonder if this can be a bit easier to understand and extend in the future if we used a string split instead.